### PR TITLE
Docker build file for sw4 enabled with hdf5 and zfp compression features

### PR DESCRIPTION
**Pull request by jf-huang**

### CODE REVIEW

**Comment by houjun** [comment]
@jf-huang I think EXTRA_CXX_FLAGS=-DACCEPT_USE_OF_DEPRECATED_PROJ_API_H should not be used, can you compile without it?

**Comment by jf-huang** [comment]
Thanks for the feedback! Yes you are right, it compiles without the EXTRA_CXX_FLAGS.

File: configs/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  make FC= CC=mpicc ZFP_HOME=/home/zfp HDF5_HOME=/usr/local/hdf5 all install && \
+  cd ..
+
+# changing BUILD_TIME will break the cache here so that sw4 can be built each time
+ARG BUILD_TIME=unknown
+
+# build sw4 
+RUN git -c http.sslVerify=false clone https://github.com/geodynamics/sw4.git && \
+  cd sw4/ && git switch developer && rm -r .git/ && \
+  ln -s /usr/lib/x86_64-linux-gnu/libssl.so.1.1  /usr/lib64/libssl.so && \
```
Comment (houjun) [inline configs/Dockerfile:34]: @jf-huang, I think the latest sw4 doesn't need libssl and libcrypto explicitly, they are needed for libproj and should link automatically. Can you try removing this and the next line to see if sw4 compiles successfully?

**Comment by houjun** [comment]
@andersp this is ready to be merged.

**Comment by andersp** [comment]
Hi Houjun,
Not sure how to merge in this PR since it comes from a forked repo, see attachment.
-Anders

From: Houjun Tang ***@***.***>
Date: Monday, March 27, 2023 at 11:20 AM
To: geodynamics/sw4 ***@***.***>
Cc: Petersson, Anders ***@***.***>, Mention ***@***.***>
Subject: Re: [geodynamics/sw4] Docker build file for sw4 enabled with hdf5 and zfp compression features (PR #117)

@andersp<https://urldefense.us/v3/__https:/github.com/andersp__;!!G2kpM7uM-TzIFchu!go9204ynnBbLXPgeNkLiASSNlTePlrrJObV6lkrKJUgRc_fvz9eSQSO9PdGDidLGpXk$> this is ready to be merged.

**Comment by houjun** [comment]
@andersp, you can just click the "Merge pull request" button if you can see it on the github pull request page.

**Comment by houjun** [comment]
@andersp  Try this link: [https://github.com/geodynamics/sw4/pull/117](https://github.com/geodynamics/sw4/pull/117)

**Pull request merged**